src/model/test/LifeCycleCostUsePriceEscalation_GTest.cpp

```diff
@@ -0,0 +1,100 @@
+*  disclaimer.
+*
+*  (2) Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+*  disclaimer in the documentation and/or other materials provided with the distribution.
+*
+*  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+
+#include "ModelFixture.hpp"
+
+#include "../LifeCycleCostUsePriceEscalation.hpp"
+#include "../LifeCycleCostUsePriceEscalation_Impl.hpp"
+
+#include "../../utilities/time/Date.hpp"
+
+using namespace openstudio;
+using namespace openstudio::model;
+
+TEST_F(ModelFixture, LifeCycleCostUsePriceEscalation) {
+
+  Model model;
+
+  LifeCycleCostUsePriceEscalation lifeCycleCostUsePriceEscalation(model);
+
+  // check defaults
+  EXPECT_TRUE(lifeCycleCostUsePriceEscalation.name());
+  EXPECT_EQ("Life Cycle Cost Use Price Escalation 1", lifeCycleCostUsePriceEscalation.name().get());
+  EXPECT_TRUE(lifeCycleCostUsePriceEscalation.resource());
+  EXPECT_EQ(lifeCycleCostUsePriceEscalation.resource().get(), "");
+  EXPECT_FALSE(lifeCycleCostUsePriceEscalation.escalationStartMonth());
+  EXPECT_FALSE(lifeCycleCostUsePriceEscalation.escalationStartYear());
+  EXPECT_EQ(lifeCycleCostUsePriceEscalation.numYears(), 0);
+
+  // check setters
+  EXPECT_TRUE(lifeCycleCostUsePriceEscalation.setResource("Electricity"));
+  EXPECT_TRUE(lifeCycleCostUsePriceEscalation.resource());
```

Comment:
(This is one of the cases where IMHO you should use "ASSERT_TRUE" or it'll crash on the next line when you do `.get()`. That's just cosmetic though, in both case you'll have a failing test)